pkg/storage/mongo/mongo.go

```diff
@@ -16,18 +16,19 @@ type ModuleStore struct {
 	url string
 }
 
-// NewStorage returns an unconnected Mongo backed storage
-// that satisfies the Backend interface.  You must call
-// Connect() on the returned store before using it.
-func NewStorage(url string) *ModuleStore {
-	return &ModuleStore{url: url}
+// NewStorage returns a connected Mongo backed storage
+// that satisfies the Backend interface.
+func NewStorage(url string) (*ModuleStore, error) {
+	ms := &ModuleStore{url: url}
+	err := ms.connect()
+	return ms, err
```

Comment:
@kteb could you return `nil, err` if the call to `connect` failed?

Reply:
:+1: